### PR TITLE
Merge npwalker/control-repo in puppetlabs/control-repo

### PR DESCRIPTION
**Pull request by npwalker**

### CODE REVIEW

**Comment by reidmv** [comment]
There's a lot going on here and I'd love to see it evolve to be simpler to get set up. That said, this is a work in progress and importantly, this code is actually being used in the real world and iterated on. Merge it, I say! Can I get another :+1: ?

**Comment by glarizza** [comment]
I'm definitely :+1: but I also asked him to do it.  It follows along the lines of what we're doing in the field, so this is great

File: README.md [hunk 1]
```diff
@@ -1,2 +1,179 @@
...
+7. Add your user to the `puppet` group as well 
+
+8. Create a project called `control-repo` and set the Namespace to be the `puppet` group
+
+9.  Logout of root and login as the `r10k_api_user`
+ - Go to profile settings -> account ( https://<your_gitlab_server>/profile/account )
+ - Copy the api token
+	
+10. Clone this control repository to your laptop/workstation
+ - `git clone https://github.com/npwalker/control-repo.git`
```
Comment (carthik) [inline README.md]: fix url to puppetlabs/

Comment (npwalker) [inline README.md]: okay added a commit to remove that url

**Comment by glarizza** [comment]
I'll give it until EOD today, if nobody dissents I'll hit the merge button.  I use this on most of my gigs, so if something arises we'll file a PR and correct it.

File: site/role/manifests/gitlab.pp [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class role::gitlab {
+
+  include profile::gitlab
```
Comment (carthik) [inline site/role/manifests/gitlab.pp]: add profile::base here, since other roles say all roles should include profile::base, perhaps?

**Comment by trlinkin** [comment]
Looking good to me! :+1: 

I would like to see a Gemfile and Rakefile for some basic checks, but thats probably another PR

**Comment by glarizza** [comment]
With nobody dissenting, I'm merging!

**Pull request merged**